### DISCUSSION TITLE
Training MACE to predict standard enthalpy of formation

### DISCUSSION THREAD

**Discussion by themonkifier** [ACEsuit/mace#717] [Q&A] [answered]
Hi all,

I have a dataset of ~10k molecules, with calculated enthalpy of formation values. As far as I can tell (I'm not much of a chemist myself), enthalpy is a form of energy that should be able to be learned and predicted by MACE. However, I'm struggling to create a good model, with all of mine plateauing at around 600-800 loss (~15000 meV energy error). So far, I've tried shifting all the enthalpy values to be less than 0 (by subtracting everything by the maximum value of the dataset), but now I'm at a loss for other possible solutions. Any suggestions are welcome.

Config:
```
name: 10k_EoF_shifted
train_file: shifted_eofs.xyz
valid_fraction: 0.10
test_file: shifted_eofs_test.xyz
E0s: average
model: MACE
num_interactions: 2
num_channels: 256
max_L: 2
correlation: 2
r_max: 5.0
forces_weight: 0
swa_forces_weight: 0
energy_weight: 10000
swa_energy_weight: 10000
lr: 0.001
swa_lr: 0.0001
energy_key: EoF
forces_key: REF_force
stress_key: REF_stress
batch_size: 2
valid_batch_size: 4
save_all_checkpoints: True
max_num_epochs: 200
start_swa: 100
scheduler_patience: 15
patience: 30
eval_interval: 4
error_table: TotalMAE
default_dtype: float64
seed: 648721
device: cuda
```

**Comment by gabor1** [comment, marked as answer]
from your input file I am guessing that you are giving forces and stresses as well, but these are not the derivatives of the enthalpy. so I would suggest that you learn the enthalpy in two steps. First you learn the atomization energy (AND DON'T USE "E0s: average", USE REAL ISOLATED ATOM E0s) , and then create a second dataset which only has "energy" values, that are the difference between your potential energy and the enthalpy that you ultimately want to learn. For learning the atomization energy, you should use the forces and stresses, because those are indeed the correct derivatives. for the second function, when you are learning the difference between energy and enthalpy, you don't have derivatives of that, so remove force and stress data (or change the label).

**Reply by gabor1** [reply]
let us know how you get on!

**Reply by themonkifier** [reply]
Thank you for the answer! I was not using forces or stresses in this case, I just gave the custom keys to suppress the warning about using the default names being incompatible with ASE. I think I understand the rest of your answer, but I do have one follow-up question. Would the E0s here be the standard enthalpies of atomization for elements in their standard state?
